src/dev/server.ts

```diff
@@ -168,6 +169,11 @@ export function createDevServer(nitro: Nitro): NitroDevServer {
     app.use(
       route,
       eventHandler(async (event) => {
+        const { rewrite } = opts;
+        if (rewrite) {
+          const path = getRequestPath(event);
+          event.node.req.url = rewrite(path);
```

Comment:
We might need to rewrite `event.path` too (if have too).

Also consider we have a built-in option for http-proxy to rewrite path as far as i remember will check.

Reply:
I just checked nodejs doc. node.req is a IncomingMessage and path is not a property of it.

Reply:
Here's what I found on http-proxy: http-proxy-middleware has a pathRewrite option, but it's not on http-proxy.